src/libmongoc/tests/unified/runner.c

```diff
@@ -1153,7 +1153,7 @@ test_check_outcome_collection (test_t *test,
       expected_sorted = bson_copy_and_sort (&expected);
 
 
-      if (!bson_equal (&actual, &expected)) {
+      if (!bson_equal (actual_sorted, expected_sorted)) {
```

Comment:
This was a small bug fix, the outcome assertions are supposed to account for different order of keys.

https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#executing-a-test

> test runners MUST permit variations in document key order or otherwise normalize the documents before comparison.